src/api/variables-api.ts

```diff
@@ -4,15 +4,31 @@ import {
 } from "../interfaces/package-manager.interfaces";
 import {httpClientV2} from "../services/http-client-service.v2";
 import {FatalError} from "../util/logger";
+import {variableAssignmentApis} from "./variable-assignment-apis";
+import {URLSearchParams} from "url";
 
 class VariablesApi {
     public static readonly INSTANCE = new VariablesApi();
+    public static readonly ASSIGNMENT_APIS  = variableAssignmentApis;
+
 
     public async assignVariableValues(packageKey: string, variablesAssignments: VariablesAssignments[]): Promise<ContentNodeTransport[]> {
         return httpClientV2.post(`/package-manager/api/nodes/by-package-key/${packageKey}/variables/values`, variablesAssignments).catch(e => {
             throw new FatalError(`Problem updating variables of package ${packageKey}: ${e}`);
         });
     }
+
+    public async getCandidateAssignments(type: string, params?: URLSearchParams): Promise<object[]> {
+        const apiUrl: string = VariablesApi.ASSIGNMENT_APIS[type].url;
+
+        if (apiUrl == null) {
+            throw new FatalError(`Variable type ${type} not supported.`);
+        }
+
+        return httpClientV2.get(apiUrl + `?${params?.toString()}`).catch(e => {
```

Comment:
In case no params are passed, then it would send `?undefined`? Does that work properly?

Reply:
nit: Not very familiar with this `httpClientV2`, but can you check if it has some other way of providing the params? Since there is a `URLSearchParams` object, I'd think is some URL object which has fields like: `url`, `params` etc.

Reply:
`httpClientV2` is a custom service that accepts only a string param for the get URL so it does not have any additional fields that we can use.